apps/hash-graph/lib/graph/src/store/test_graph.rs

```diff
@@ -0,0 +1,146 @@
+
+impl From<Entity> for EntityRecord {
+    fn from(entity: Entity) -> Self {
+        Self {
+            properties: entity.properties,
+            metadata: EntityMetadata {
+                record_id: entity.metadata.record_id(),
+                entity_type_id: entity.metadata.entity_type_id().clone(),
+                temporal_versioning: Some(entity.metadata.temporal_versioning().clone()),
+                custom: CustomEntityMetadata {
+                    provenance: Some(entity.metadata.provenance()),
+                    archived: Some(entity.metadata.archived()),
+                },
+            },
+            link_data: entity.link_data,
+        }
+    }
+}
+
+#[derive(Debug, Default, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CustomOntologyMetadata {
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub provenance: Option<ProvenanceMetadata>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub temporal_versioning: Option<OntologyTemporalMetadata>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub owned_by_id: Option<OwnedById>,
+    #[serde(
+        default,
+        skip_serializing_if = "Option::is_none",
+        with = "time::serde::iso8601::option"
+    )]
+    pub fetched_at: Option<OffsetDateTime>,
+}
+
+impl CustomOntologyMetadata {
+    #[must_use]
+    const fn is_empty(&self) -> bool {
+        self.provenance.is_none()
+            && self.temporal_versioning.is_none()
+            && self.owned_by_id.is_none()
+            && self.fetched_at.is_none()
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct OntologyTypeMetadata {
+    pub record_id: OntologyTypeRecordId,
+    #[serde(default, skip_serializing_if = "CustomOntologyMetadata::is_empty")]
+    pub custom: CustomOntologyMetadata,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(
+    rename_all = "camelCase",
+    bound(
+        serialize = "T::Representation: Serialize",
+        deserialize = "T::Representation: Deserialize<'de>"
+    )
+)]
+pub struct OntologyTypeRecord<T: OntologyType> {
+    pub schema: T::Representation,
+    pub metadata: OntologyTypeMetadata,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct OntologyTemporalMetadata {
+    pub transaction_time: Option<LeftClosedTemporalInterval<TransactionTime>>,
+}
+
+#[derive(Debug, Default, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CustomGlobalMetadata;
+
+impl CustomGlobalMetadata {
+    #[must_use]
+    #[expect(clippy::unused_self)]
+    const fn is_empty(&self) -> bool {
+        true
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct TestData {
+    pub block_protocol_spec: semver::Version,
```

Comment:
I'd potentially change this to a struct, perhaps something like
```rust
struct BlockProtocolModuleVersions {
  graph: Option<semver::Version>
}
```

which would make it forwards compatible if we ever had other modules we needed to consider, e.g.

```rust
struct BlockProtocolModuleVersions {
  graph: Option<semver::Version>,
  authorization: Option<semver::Version>
}
```

Maybe it doesn't make sense to have graph as `Option` at the moment though, I guess we could add that in the future without it breaking previous serialized formats

Reply:
Why would you make these optional?

Reply:
> Why would you make these optional?

So that it's forwards compatible.

Reply:
^ can we do this then?

Reply:
Yes, I'm about to do this now, just didn't got to it, yet 😄

Reply:
No worries, only asked because my review was re-requested